lms/static/my.js

```diff
@@ -16,6 +18,38 @@ function escapeUnicode(str) {
 }
 
 
+function shareSolution(solutionId, isShared, sharedByUser, button, getLinkButton) {
+  const element = document.getElementById('share-solution');
+  button.style.display = (isShared == 'True') ? 'block' : 'none';
+  getLinkButton.style.display = (sharedByUser == 'True') ? 'block' : 'none';
+  button.addEventListener('click', () => {
+    const xhr = new XMLHttpRequest();
+    xhr.open('POST', '/share');
+    xhr.setRequestHeader('Content-Type', 'application/json');
+    xhr.responseType = 'json';
+    xhr.onreadystatechange = () => {
+      if (xhr.readyState === 4) {
+        if (xhr.status === 200) {
+          if (xhr.response.shared) {
+            element.style['color'] = clickedColor;
+            getLinkButton.style.display = 'block';
```

Comment:
Prefer using CSS and adding/removing classes or data attributes.

Reply:
Didn't understand the problem here